### PR TITLE
Added support for tlsConfig for ServiceMonitor

### PR DESCRIPTION
**Pull request by mprimeaux**
This PR addresses #180. 

- Added `tlsConfig` within the `ServiceMonitor` stanza.
- Extended servicemonitor.yaml template to include `tlsConfig` if defined.

The [CONTRIBUTING.md](https://github.com/cockroachdb/helm-charts/blob/master/cockroachdb/CONTRIBUTING.md) instructions indicated to increment the version within the `Chart.yaml`. However, the first line in `Chart.yaml` indicates the file is generated and to see [/build/templates/Chart.yaml](https://github.com/cockroachdb/helm-charts/blob/master/build/templates/Chart.yaml). 

If you'd like me to increment the version, then please let me know how best to proceed.  Happy to help.

### CODE REVIEW

File: build/templates/README.md [hunk 1]
```diff
@@ -367,6 +367,7 @@ For details see the [`values.yaml`](values.yaml) file.
 | `serviceMonitor.annotations`                              | Additional annotations of ServiceMonitor                        | `{}`                                                  |
 | `serviceMonitor.interval`                                 | ServiceMonitor scrape metrics interval                          | `10s`                                                 |
 | `serviceMonitor.scrapeTimeout`                            | ServiceMonitor scrape timeout                                   | `nil`                                                 |
+| `serviceMonitor.tlsConfig`                                | Additional TLS configuration of ServiceMonitor                  | `{}`                                                  |
```
Comment (prafull01) [inline build/templates/README.md:370]: You must add this change in the cockroachdb/README.md and README.md file as well

File: build/templates/values.yaml [hunk 2]
```diff
@@ -349,6 +349,10 @@ serviceMonitor:
   # Limits the ServiceMonitor to the current namespace if set to `true`.
   namespaced: false
 
+  # tlsConfig: TLS configuration to use when scraping the endpoint.
+  # Of type: https://github.com/coreos/prometheus-operator/blob/main/Documentation/api.md#tlsconfig
+  tlsConfig: {}
+
```
Comment (prafull01) [inline build/templates/values.yaml:355]: You have to add this change in the cockroachdb/values.yaml file as well to make the tests pass.

Comment (mprimeaux) [reply build/templates/values.yaml:355]: Thank you, @prafull01!

File: build/templates/README.md (hunk 1 above)
Comment (prafull01) [reply build/templates/README.md:370]: @mprimeaux You missed this one.

**Comment by mprimeaux** [comment]
Looks like the failures are due to the Docker container registry having issues, [here](https://github.com/cockroachdb/helm-charts/actions/runs/6340264721/job/17231368764?pr=353#step:4:62) and [here](https://github.com/cockroachdb/helm-charts/actions/runs/6340264721/job/17231369012?pr=353#step:4:65).

**Comment by mprimeaux** [comment]
@prafull01 Looks like the yesterday's Docker container registry service disruption caused the build to fail. Would you mind please kicking it off again? Thanks much in advance.

Also, before doing that, do I need to increment the minor version of the chart or is this handled automatically by the build process?

**Comment by mprimeaux** [comment]
@prafull01 Just a friendly nudge in case this has dropped off your "list" :)  Again, continued thanks for your support and help.

**Review by prafull01 — APPROVED** [review]

**Pull request merged**